src/config.rs

```diff
@@ -511,20 +511,69 @@ impl<'a> Cfg<'a> {
                 )))
             })
             .transpose()?;
-        self.local_toolchain(toolchain)
+        self.local_toolchain(toolchain).await
     }
 
-    pub(crate) fn find_active_toolchain(
+    pub(crate) async fn find_active_toolchain(
         &self,
+        force_install_active: Option<bool>,
     ) -> Result<Option<(LocalToolchainName, ActiveReason)>> {
-        Ok(
-            if let Some((override_config, reason)) = self.find_override_config()? {
+        Ok(match self.find_override_config()? {
+            Some((
+                OverrideCfg::Official {
+                    components,
+                    targets,
+                    profile,
+                    ref toolchain,
+                },
+                reason,
+            )) => {
+                let should_install_active = force_install_active.unwrap_or_else(|| {
+                    self.process
+                        .var("RUSTUP_AUTO_INSTALL")
+                        .map_or(true, |it| it != "0")
+                        || self.process.var("CI").is_ok() && self.process.var("RUSTUP_CI").is_err()
```

Comment:
```suggestion
```

Let's not do this for now. The `RUSTUP_AUTO_INSTALL` env being set to `true` if missing is enough for the initial fix and in any case the env, if set, should probably override any implicit behaviours.